src/test/java/org/javarosa/xform/parse/XFormParserTest.java

```diff
@@ -130,6 +132,19 @@ public void parsesPreloadForm() throws IOException {
         assertEquals("Sample Form - Preloading", formDef.getTitle());
     }
 
+    @Test
+    public void parsesExternalSecondaryInstanceAsFormInstance() throws IOException {
+        Path form = r("Sample-Preloading.xml");
```

Comment:
It would be ideal to have a test that just isolates the case that is trying to be handled here. For example, a very narrow one could define an external CSV instance that is not referred to in an `instance()` function. A slightly broader test form would have one that is and one that isn't so both of those cases can be verified.